### ISSUE TITLE
no-unused-expressions makes an error for optional chaining of method calls: foo?.bar();

### ISSUE THREAD

**Issue by finom** [eslint/eslint#11045]
![image](https://user-images.githubusercontent.com/1082083/47850435-e8ea9a00-dddd-11e8-8d3d-9a5b891ad212.png)

**Tell us about your environment**

* **ESLint Version:** 5.8.0
* **Node Version:** 10.1.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

<details>
<summary>
The config
</summary>

```js
module.exports = {
  extends: 'airbnb-base',
  parser: 'babel-eslint',
  plugins: ['babel'],
  rules: {
    'import/prefer-default-export': 0,
    'no-plusplus': 0,
    'no-underscore-dangle': 0,
    'no-mixed-operators': 0,
    'no-prototype-builtins': 0,
    'no-continue': 0,
    'no-param-reassign': ['error', { props: false }],
    'import/no-extraneous-dependencies': 0,
    'class-methods-use-this': 0,
  },
  globals: {
    window: true,
    document: true,
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
foo?.bar();
```

```bash
eslint 'src/**/*.js' 'src2/**/*.js' '*.js' --ignore-pattern '!.eslintrc.js'
```

**What did you expect to happen?**

I expect `no-unused-expressions` rule to not make the error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
```

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by platinumazure** [comment]
Apologies @finom for letting this slip through the cracks. @ronen has the right idea-- core ESLint doesn't support this, so you need to use the eslint-plugin-babel rule.

Closing as this is currently unsupported. If optional chaining becomes a Stage 4 ECMAScript proposal, we could reopen this and make any necessary changes to the core rule. Thanks!

**Comment by finom** [comment]
@ronen @platinumazure thank you guys.
